lib/scss-preprocessor/index.js

```diff
@@ -1,23 +1,55 @@
 /* jshint node: true */
 'use strict';
 
+var path = require('path');
+
 var funnel = require('broccoli-funnel');
 var mergeTrees = require('broccoli-merge-trees');
 
-var WriteMixins = require('./broccoli-component-mixins');
-var WriteComponentImports = require('./broccoli-auto-import-components');
+var TransformComponents = require('./broccoli-transform-components');
+var ImportThemeComponents = require('./broccoli-import-theme-components');
+var ProvideAppImports = require('./broccoli-provide-app-imports');
 
 module.exports = function(tree, options) {
-  var components = options.components;
+  var themes = options.themes || [];
+  var moduleNamespace = options.moduleNamespace;
+  var stylesNamespace = options.stylesNamespace;
+  var componentsNamespace = options.componentsNamespace;
+
+  var configFileName = 'config.scss';
+  var scssMainFilePath = 'app/styles/app.scss';
+
+  var themeTrees = themes.reduce(function(trees, theme) {
+    var themeTree = theme.toScssTree();
+    themeTree = funnel(themeTree, {
+      destDir: path.join(moduleNamespace, theme.name, stylesNamespace)
+    });
+    themeTree = new TransformComponents(themeTree, {
+      // modules/ui-base-theme/styles/components/ui-button--default.scss => match "ui-base-theme"
+      themePattern: new RegExp('^'+moduleNamespace+'\/('+theme.name+')\/'+stylesNamespace+'\/'+componentsNamespace+'\/.+\.scss$'),
+      // modules/ui-base-theme/styles/components/ui-button--default.scss => match "ui-button--default"
+      namePattern: new RegExp('^'+moduleNamespace+'\/.+\/'+stylesNamespace+'\/'+componentsNamespace+'\/(.+)\.scss$')
+    });
+
+    var themeMain = path.join(moduleNamespace, theme.name, stylesNamespace, theme.name + '.scss');
+    themeTree = new ImportThemeComponents(themeTree, {
```

Comment:
Step 2: Generate the "main" file for our theme (i.e.: `ui-base-theme.scss`) which auto-imports our theme's component scss files